### ISSUE TITLE
Occasional linux test failure with hypothesis created data.

### ISSUE THREAD

**Issue by charris** [numpy/numpy#27422]
See https://github.com/numpy/numpy/actions/runs/10924770357/job/30325032749.
The failure is in `test_array_scalar_ufunc_equivalence` and may, or may not, be related to #27418. @jorenham Thoughts?

**Comment by jorenham** [comment]
#27418 only affected a `.pyi` stub, and I believe that this failing test is a runtime one, so I don't think that there's a causal relation

**Comment by jorenham** [comment]
FWIW. this sounds like a classic case of flakiness to me:

```
| AssertionError: 
| Arrays are not equal
| 
| Mismatched elements: 1 / 1 (100%)
| Max absolute difference among violations: 5.9604645e-08
| Max relative difference among violations: 2.2240386e-08
|  ACTUAL: array(0.65+2.6j, dtype=complex64)
|  DESIRED: array(0.65+2.6j, dtype=complex64)
```

Maybe some recent event caused the hypothesis cache to be invalidated?

**Comment by seberg** [comment]
The first error may be related to the FMA work/tests, or maybe just runner changes themselves using FMA/not using FMA for complex multiplication.  (It seems indeed harmless.)
There is also a datetime failure comparing a complex with a `NaT` datetime.  It isn't quite clear what happened there.  There is an `xfail` for that so it may just be that hypothesis reports it incorrectly.

**Comment by charris** [comment]
Same at https://github.com/numpy/numpy/actions/runs/11110913872/job/30870108510?pr=27486

**Comment by seberg** [comment]
Yeah, same thing.  Still think all of these are just fine, but I am not sure hypothesis likes the `xfail`.
I suspect using `almost_equal_*` may just create it's own set of problems, so I am somewhat tempted to do what the comment of the test says: "If this creates problems, maybe just delete it..."

(alternatively, we can look into not only sampling a few dtypes.)

**Comment by mattip** [comment]
cc @honno as a hypothesis/pytest expert, can you suggest anything?

**Comment by charris** [comment]
Another case at https://github.com/numpy/numpy/actions/runs/14505328602/job/40693911668. Checking by hand, the expected value is correct and the returned value differs by 1 ulp. However, the result is in the range where there are missing integers,  so this can probably be regarded as floating point error. The Hypothesis generated test data should probably be adjusted or results checked within an ulp of two.

**Comment by charris** [comment]
And another https://github.com/numpy/numpy/actions/runs/14541363659/job/40800139903?pr=28771. These are no longer rare, we might want to look into them.
